keps/sig-release/20200122-kubernetes-yearly-support-period.md

```diff
@@ -0,0 +1,318 @@
+With a new minor release coming every three months, there is a constant need to evaluate and plan and enact forward movement.
+
+The survey conducted in early 2019 by the WG LTS showed that a significant subset of Kubernetes end-users fail to upgrade within the 9-month support period.
+In the graph below, 1.13 was the current version, meaning that 1.9 and 1.10 were a few months out of support.
+Yet about 1/3 of users had production machines on those versions:
+
+![Kubernetes versions in production](./20200122-versions-in-production.png)
+
+This, and other responses from the survey, suggest that this 30% of users would better be able to keep their deployments on supported versions if the patch support period were extended to 12-14 months.
+This appears to be true regardless of whether the users are on DIY build or commercially vendored distributions.
+An extension would thus lead to more than 80% of users being on supported versions, instead of the 50-60% we have now.
+
+A yearly support period provides the cushion end-users appear to desire, and is more in harmony with familiar annual planning cycles.
+
+We view elongating the patch support window toward 1 year as a relatively minor change, yet is believed viable from an implementation perspective and is a meaningful first response to users.
+
+There are many unknowns about changing the support windows for a project with as many moving parts as Kubernetes, and keeping the change relatively small (relatively being the important word), gives us the chance to find out what those unknowns are in detail and address them.
+
+### Goals
+
+- Address end-user feedback regarding support time per release
+- Define an annual support policy.
+- Define any prerequisites for the policy.
+- Set acceptance criteria for enacting the change in support policy. This should be supplemented by KEPs and PRs that address the implementation of the policy by the release team and Release Engineering (patch management).
+- Clarify and confirm that the support period is defined based on time (e.g. 1 year) and not based on a specific number of releases (e.g. 3 or 4) given that the release cadence might change in the future.
+
+### Non-Goals
+
+- Define all details of “support”, which is somewhat imprecise in the project already today.
+- Change the release cadence.  It is currently 3 months per release or 4 releases per year.  See “Alternatives” below for related work on release cadence.
+- Establish an “LTS” release process in the sense perhaps known from other projects (e.g.: pick one release every year or two, give patch support on that release for multiple years).
+- Declare that Kubernetes’ dependencies must also have a similar support lifetime per release.
+- Align Kubernetes releases in time with those of its dependencies.
```

Comment:
we do need to consider what we will do if there is a security related issue in a dependency that is only fixed in versions that are significant/risky to update to in older kubernetes releases. To be clear, we already have this issue occasionally, but adding another kubernetes release makes it more likely. outlining how we currently align with our biggest dependencies (go, etcd, cadvisor, cloud provider libraries, etc) would be helpful in understanding if/how this proposal would impact current alignment.

Reply:
I agree that needs to be done, but I'm not sure how to fold it in here.

Reply:
agreed; but the most we can do is have a clearer policy for this.  It's already a problem.